appinventor/appengine/src/com/google/appinventor/client/Ode.java

```diff
@@ -2344,6 +2353,20 @@ public boolean getDeleteAccountAllowed() {
     return config.getDeleteAccountAllowed();
   }
 
+  public static void setupOrigin(ServiceDefTarget service) {
```

Comment:
To discuss: In the branch I have for doing local services, the local versions obviously aren't subclasses of `ServiceDefTarget` so all of the casts added to the code will fail. I'd like to consider changing the signature of this function to take `Object` and then check to see if `service instanceof ServiceDefTarget` to centralize the type checking logic here. Obviously, if the service isn't a `ServiceDefTarget` (i.e., it's the local version) we just do nothing.

Reply:
I made a draft PR #3147 to show how the local services work and the change I made to the RpcStatusPopup to handle local vs remote service definitions.